server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -305,13 +358,31 @@ boolean uploadNewSegments(Collection<String> localFiles) throws IOException {
                 );
                 return true;
             }
-        }).forEach(file -> {
+        }).collect(Collectors.toList());
+
+        // Start tracking the upload bytes started
+        filesToUpload.forEach(file -> segmentTracker.addUploadBytesStarted(sizeMap.get(file)));
```

Comment:
Would it help if we start tracking segment file specific metrics from `RemoteSegmentStoreDirectory`?

Reply:
We can do that, but it would lead to scattered code for the tracker updation.

Reply:
have refactored the code, pls check now.